### DISCUSSION TITLE
What configurations could help with filer memory usage?

### DISCUSSION THREAD

**Discussion by YixingCheng** [seaweedfs/seaweedfs#6012] [Q&A] [answered]
Hi community, we are facing an OOMKill issue on the filer/s3 server due to a large spike of upload requests from a seaweed backed registry. So we could expect the the updated files could be anywhere from xMB to xGB.
We are using the default local leveldb2 as the filer store.

We can see the memory usage correlates to the requests rate.
![image](https://github.com/user-attachments/assets/25562cbe-a5e8-449c-8ed4-004494838212)
![image](https://github.com/user-attachments/assets/2cde5dc9-3a29-4a07-8480-0cf9d1c15423)

I know that we could do something like rate limiting or throttling in case of such large spike, but I want to ask if there are some filer configurations could help with the memory usage in the filer.
Will lower the default `concurrentUploadLimitMB` help in this case?

Thanks for your inputs.

**Comment by chrislusf** [comment, marked as answer]
* Yes
* Use a shared filer store and increase the number of filer/s3 instances.
